### DISCUSSION TITLE
Imported realm user federation settings not in database tables

### DISCUSSION THREAD

**Discussion by fostermi** [keycloak/keycloak#26754] [Q&A]
We recently upgrade to Keycloak.X and did a full realm import and everything seems to be working fine, including our LDAP user federation. However, we moved our LDAP server, updated DNS and suddenly KC couldn't find the LDAP server at the new location. It was still trying to use the old location. In trying to troubleshoot why KC was using the old server address, I looked in the database tables to see how it was configured and couldn't find any user federation configured in any table with "user" or "federation" at all. Is there a reason why our running configuration doesn't seem to be in the database at all?

**Comment by darius-m** [comment]
User federations are not defined in the federation tables (I think "federation" in this context refers to SSO federation, while LDAP / AD are user databases). You should look into the `COMPONENT` and `COMPONENT_CONFIG` tables for LDAP configuration. If you have database access you can also find where data is stored by looking for known strings in a dump of the database or using a database explorer.

**Comment by fostermi** [comment]
Ah, thank you. The LDAP config is indeed in the `COMPONENT_CONFIG` table.
